packages/shared/lib/sdk/sync.ts

```diff
@@ -443,11 +435,8 @@ export class NangoAction {
 
     public ActionError = ActionError;
 
-    private memoizedConnections: Map<string, { connection: Connection; timestamp: number } | undefined> = new Map<
-        string,
-        { connection: Connection; timestamp: number }
-    >();
-    private memoizedIntegration: GetPublicIntegration['Success']['data'] | undefined;
+    private memoizedConnections = new Map<string, { connection: Connection; timestamp: number }>();
+    private memoizedIntegration = new Map<string, { integration: GetPublicIntegration['Success']['data']; timestamp: number }>();
```

Comment:
Just curious here - what's the use case around needing to access any multiple integrations?

Reply:
It's not the case here, the key is the sum of query params which can change the shape so it's going to store different shape of the same integration depending on the params. Not ideal (cf PR desc)